sdk/core/Azure.Core/src/AsyncCollection.cs

```diff
@@ -13,7 +13,7 @@ namespace Azure
     /// iterate over.
     /// </summary>
     /// <typeparam name="T">The type of the values.</typeparam>
-    public abstract class AsyncCollection<T> : IAsyncEnumerable<Response<T>> where T : notnull
+    public abstract class AsyncCollection<T> : IAsyncEnumerable<T> where T : notnull
```

Comment:
Do we care about duck typing support and providing method that will return struct enumerator for this collection, similar to `ConfiguredCancelableAsyncEnumerable`?

Reply:
I don't think we can return struct iterator considering this has to be an overrideable abstraction.